docs/design/generic.md

```diff
@@ -0,0 +1,126 @@
+# Generic artifact fetching
```

Comment:
nitpick: I don't think `generic.md` is a self-descriptive name and gives way too much room for imagination without reading the design - maybe `generic_fetcher.md` might sound a little better.

Reply:
Renamed